src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -61,11 +61,11 @@ public RoslynAnalyzerProvider(IAnalyzerInstaller analyzerInstaller, ILogger logg
         /// Active rules should never be empty, but depending on the server settings of repo keys, we might have no rules in the ruleset.
         /// In that case, this method returns null.
         /// </summary>
-        public AnalyzerSettings SetupAnalyzer(TeamBuildSettings teamBuildSettings, IDictionary<string, string> serverProperties,
+        public AnalyzerSettings SetupAnalyzer(TeamBuildSettings teamBuildSettings, IAnalysisPropertyProvider sonarProperties,
```

Comment:
Similar to the previous commit - this class should use the analysis property provider interface, rather than a simple map of values.